pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -263,8 +264,7 @@ public synchronized void readMoreEntries() {
                             consumerList.size());
                 }
                 havePendingRead = true;
-                cursor.asyncReadEntriesOrWait(messagesToRead, serviceConfig.getDispatcherMaxReadSizeBytes(),
-                        this,
+                cursor.asyncReadEntriesOrWait(messagesToRead, bytesToRead, this,
```

Comment:
bytesToRead can potentially be -1 now (like at line 315 when `new MutablePair<>(-1, -1);` is returned) when dispatch limit is not set. This changes the default behaviour because earlier it was capped by `serviceConfig.getDispatcherMaxReadSizeBytes()`